app/src/view/DatePicker.js

```diff
@@ -0,0 +1,161 @@
+      paddingTop: '6px',
+      marginTop: '16px',
+      marginBottom: '8px',
+      width: '100%',
+      '& .react-datepicker-wrapper': {
+        display: 'inline',
+      },
+    },
+    datepicker: {
+      backgroundColor: theme.palette.background.paper,
+      borderColor: borderColor,
+      borderTopColor: 'transparent',
+      border: 'solid 1px',
+      borderRadius: theme.shape.borderRadius,
+      padding: '18.5px 14px',
+      color: theme.palette.text.primary,
+      fontSize: theme.typography.fontSize,
+      lineHeight: '18px',
+      width: '100%',
+      '&:hover': {
+        borderColor: theme.palette.text.primary,
+        borderTopColor: 'transparent',
+      },
+    },
+    span: {
+      position: 'absolute',
+      top: '0',
+      left: '0',
+      display: 'flex',
+      width: '100%',
+      fontSize: '10px',
+      lineHeight: '15px',
+      '&:before,&:after': {
+        content: '""',
+        display: 'block',
+        boxSizing: 'border-box',
+        marginTop: '6px',
+        borderTop: 'solid 1px',
+        borderColor: borderColor,
+        minWidth: '10px',
+        height: '8px',
+      },
+      '&:hover::before,&:hover::after': {
+        borderColor: theme.palette.text.primary,
+      },
+      '&:before': {
+        marginRight: '4px',
+        borderLeft: 'solid 1px transparent',
+        borderRadius: '4px 0',
+      },
+      '&:after': {
+        flexGrow: '1',
+        marginLeft: '4px',
+        borderRight: 'solid 1px transparent',
+        borderRadius: '0 4px 4px 0',
+        height: '57px',
+      },
+    },
+    '@global': {
+      '#datepickers .react-datepicker__header, #datepickers .react-datepicker__current-month, #datepickers .react-datepicker__day-name, #datepickers .react-datepicker-time__header': {
+        backgroundColor: theme.palette.background.default,
+        color: theme.palette.text.primary,
+        borderColor: borderColor,
+      },
+      '#datepickers .react-datepicker, #datepickers .react-datepicker__day, #datepickers .react-datepicker__time, #datepickers .react-datepicker__time-container': {
+        backgroundColor: theme.palette.background.paper,
+        color: theme.palette.text.primary,
+        borderColor: borderColor,
+      },
+      '#datepickers .react-datepicker__time-list-item': {
+        height: 'auto',
+        padding: '7px 10px',
+      },
+      '#datepickers .react-datepicker__day--selected, #datepickers .react-datepicker__time-list-item--selected': {
+        backgroundColor: theme.palette.primary.main,
+        color: theme.palette.primary.contrastText,
+      },
+      '#datepickers .react-datepicker__day:hover, #datepickers .react-datepicker__time-list-item:hover': {
+        backgroundColor: theme.palette.background.default,
+      },
+    },
+  };
+});
+
+const Portal = ({children}) => {
+  const portalRoot = document.getElementById("portal");
+  const el = document.createElement("div");
+  el.id = "datepickers";
+
+  if (!portalRoot) {
```

Comment:
I added this section to deal with 2 flow errors. I'm not sure if there's a better way of handling this, but my thinking is that in the unlikely event that `#portal` can't be found in the wild, at least we'll have an error in the console.

```
 Error ------------------------------------------------------------------------------------- src/view/DatePicker.js:92:16

Cannot call `portalRoot.appendChild` because property `appendChild` is missing in null [1].

   src/view/DatePicker.js:92:16
     92|     portalRoot.appendChild(el);
                        ^^^^^^^^^^^

References:
   /tmp/flow/flowlib_e4a0f5a/dom.js:1250:52
   1250|   getElementById(elementId: string): HTMLElement | null;
                                                            ^^^^ [1]


Error ------------------------------------------------------------------------------------- src/view/DatePicker.js:93:29

Cannot call `portalRoot.removeChild` because property `removeChild` is missing in null [1].

   src/view/DatePicker.js:93:29
     93|     return () => portalRoot.removeChild(el);
                                     ^^^^^^^^^^^

References:
   /tmp/flow/flowlib_e4a0f5a/dom.js:1250:52
   1250|   getElementById(elementId: string): HTMLElement | null;
                                                            ^^^^ [1]

```

Reply:
Yup, this is the right way to do it.